adr/ADR-40.md

```diff
@@ -0,0 +1,49 @@
+# Request Many
+
+| Metadata | Value                 |
+|----------|-----------------------|
+| Date     | 2023-06-26            |
+| Author   | @aricart, @scottf     |
+| Status   | Partially Implemented |
+| Tags     | client                |
+
+## Problem Statement
+Have the client support receiving multiple replies from a single request, instead of limiting the client to the first reply.
+
+## Design
+
+Making the request and handling multiple replies is straightforward. Much like a simplified fetch, the developer 
+will provide some basic strategy information that tell how many messages to wait for and how long to wait for those messages.
+
+## Options
```

Comment:
Do we want to have some common defaults across clients around max time or gap time?

Reply:
we can but reallly that value should be based on the worse rtt for the furthest service

Reply:
Timeout shouldn't necessarily be dictated by RTT (although tbh it is in most cases).  There are cases where there's value in determining that SLAs cannot be met and a hard timeout would indicate that - particularly around time dependent data where business value decreases over time or may even be harmful/misleading if stale.